pallets/liquidity-pools/src/message.rs

```diff
@@ -79,7 +200,7 @@ impl TryInto<Domain> for SerializableDomain {
 	TypeInfo,
 	MaxEncodedLen,
 )]
-pub enum Message {
+pub enum Message<BatchContent = BatchMessages> {
```

Comment:
This allows us to create different versions of Message where we can stop the recursive process of `max_encoded_len()`. By default, we want the `Message` to be able to batch another messages. Except when the message already belongs to a batch.

Reply:
Not sure I am following ^^

Reply:
The problem: The worst case for `MaxEncodedLen` autoimplementation is that the `Message` contains a `Batch` that contains `Messages` that contain `Batch` that contains `Messages`...

Parameterizing here with the batch content we are able to create a `Message<()>` to compute correctly the `MaxEncodedLen` for cases where the message is a submessage of a batch, and avoid the infinite recursion